audio-worklet/bitcrusher-test.html

```diff
@@ -16,171 +16,184 @@
-    	  const reduction = 7; 
-    	  const bitDepth = 24;
-    	  const bufferSize = 512;
-    	  let oscillator = new OscillatorNode(context);
-    	  let splitter = new ChannelSplitterNode(context);
-
-    	  let bitcrusher = new Bitcrusher(context, {
-    	    bufferSize: bufferSize,
-    	    channels: channels,
-    	    bitDepth: bitDepth,
-    	    reduction: reduction
-    	  })
-
-    	  oscillator.connect(splitter).connect(bitcrusher.input);
-    	  bitcrusher.output.connect(context.destination); 
-    	  oscillator.start();
-    	  
-    	  // When audio buffer is ready, verify bitcrushed samples are unaltered.
-    	  context.startRendering().then((buffer) => {      
-    	      let last = buffer.getChannelData(0)[0];
-    	      for (let i = 1; i < buffer.getChannelData(0).length; i++) {
-    	        for (let c = 0; c < channels; c++) {
-    	          console.assert(
-    	              buffer.getChannelData(c)[i] === last, 
-    	              'At iteration ' + i + ', the value at channel ' + c + ', ' +
-    	                  buffer.getChannelData(c)[i] + ' did not match ' +
-    	                  ' expectation, ' + last);
-    	          if (i % reduction === 0) last = buffer.getChannelData(c)[i];
-    	        }
-    	     } 
-    	  });  
-    	}
+      testSampleReduction();
+      testBitcrusherInAudioContext();
+      testMultipleChannels();
+      testIllegalInput();
+
+      /**
+       * Populate array with 0:99 and verify the bitcrusher holds an array
+       * where the number of consecutive equal numbers is controlled by the 
+       * variable factor. This verfies the sample rate reduction component
+       * of bitcrusher operating outside of an audio context.
+       */
+      function testSampleReduction() {
+        let context = new OfflineAudioContext(1, 44100, 44100);
+
+        for (let reduction = 1; reduction < 10; reduction++) {
+          const bufferSize = 5;
+          let referenceBuffer = new Float32Array(bufferSize);
+          let outputBuffer = new Float32Array(bufferSize);
+          let bitcrusher =
+              new Bitcrusher(context, {bitDepth: 24, reduction: reduction});
+
+          for (let i = 0; i < bufferSize; i++) {
+            referenceBuffer[i] = i;
+          }
+          bitcrusher.processBuffer_(referenceBuffer, outputBuffer, 0);
+          
+          // Verify computed values match expected values.
+          outputBuffer.map((data, index) => {
+            let expected = Math.floor(index / reduction) * reduction;
+            console.assert(
+                data === expected, 
+                'computed ' + data + ' but expected ' + expected);
+          });
+        }
+      }
+
+      /**
+       * Create one oscillator, and connect it to a bitcrusher with variables 
+       * set to avoid any effect and a delay node which compensates for the 
+       * script processor latency. Then verify that the samples are nearly
```

Comment:
Having this description closer to where you create the graph would be better so that code hs comments for the purpose of the graph.